src/coreclr/src/System.Private.CoreLib/src/System/TypedReference.cs

```diff
@@ -7,15 +7,16 @@
 
 using System.Reflection;
 using System.Runtime.CompilerServices;
+using Internal.Runtime.CompilerServices;
 
 namespace System
 {
     [CLSCompliant(false)]
     [System.Runtime.Versioning.NonVersionable] // This only applies to field layout
     public ref struct TypedReference
     {
-        private IntPtr Value;
-        private IntPtr Type;
+        private readonly ByReference<byte> _value;
```

Comment:
This is a breaking change.

`ByReference<T>` is classified as a "managed" type, so anything using `TypedReference` as a pointer or as part of the `unmanaged` constraint will no longer compiler after this goes in.

Reply:
Stopping people from doing https://twitter.com/korifey_ad/status/1169927122959392769 would be a nice side effect (also another point for `unmanaged` being a pointless feature).

We don't have to expose this in the reference assemblies though.

Reply:
Actually, this might be ok...
It looks like the compiler special cases `Span<T>` and `ReadOnlySpan<T>` explicitly, rather than `ByReference<T>` (likely due to it being internal).